modules/porous_flow/include/userobjects/PorousFlowDictator.h

```diff
@@ -148,6 +148,18 @@ class PorousFlowDictator : public GeneralUserObject, public Coupleable
    */
   FEType feType() const;
 
+  /**
+   * Check if the simulation includes derivatives of permeability
+   * Note: when the permeability is constant, expensive tensor calculations
+   * can be ignored in Jacobian calculations
+   */
+  bool usePermDerivs() const { return _perm_derivs; };
+
+  /**
+   * Set the _perm_derivs flag
+   */
+  void usePermDerivs(bool flag) const { _perm_derivs = flag; };
```

Comment:
This whole concept is super nice.  It doesn't make the code crazily complicated, but offers nice speedups.

I really don't the way you've got `usePermDerivs` with multiple signatures!  Quite confusing to my mind, but obviously no one else cared :-)

Reply:
Haha - I started like this, changed it to different function names but as they were quite similar I couldn't remember which was which, so I changed it back